### ISSUE TITLE
Grid object output is a bit confusing...

### ISSUE THREAD

**Issue by stb2145** [xgcm/xgcm#307]
This is a small issue regarding the output of a Grid object. I ran:
```
grid = xgcm.Grid(ds_wg, coords={'X': {'center': 'lon', 'right': 'lon_u'}, 
                                'Y': {'center': 'lat', 'right': 'lat_u'},
                                'T': {'center': 'time'},
                                'Z': {'center': 'level', 'right': 'level_w'}},
                        periodic=['X'])
grid
```
The output,
```
<xgcm.Grid>
X Axis (periodic, boundary=None):
  * center   lon --> right
  * right    lon_u --> center
Y Axis (not periodic, boundary=None):
  * center   lat --> right
  * right    lat_u --> center
T Axis (not periodic, boundary=None):
  * center   time
Z Axis (not periodic, boundary=None):
  * center   level --> right
  * right    level_w --> center
```
 is a bit confusing to me because for people unfamiliar with `xgcm.Grid()`, such as myself, I misinterpret the output as mapping the lon/lon_u, lat/lat_u, level/level_w coords as the opposite of what I specified when creating the Grid object. This might just be a personal issue, but if there are others who have also misinterpreted the output upon first glance, it may be worth adding additional notes in the output to help users understand what exactly xgcm did.

**Comment by jbusecke** [comment]
Hi @stb2145 , thanks for using xgcm and raising this issue. I have heard this comment from several other users and think it is worth improving. This is one of the first things that users see when they interact with xgcm, and it should be as clear as possible.

In the long term, we could probably weave a more intuitive representation into something like #276, but we also need a clear text representation.

Some alternate suggestions that just came to my mind:
```
<xgcm.Grid>
X Axis (periodic, boundary=None):
  * `lon` [position: center] --> right
  * `lon_u` [position: right] --> center
```
would have the same information, but more centered on the name of the dimension. To me this is more clear about where the variable 'is right now'.

Any other suggestions? Happy to implement this once we all agree that it is worthwhile and how to format this.

**Comment by rabernat** [comment]
The ` --> right` indicates the "default" shift. I would favor just removing this part from the repr.

**Comment by jbusecke** [comment]
I think it is useful to have that available though. Should we make a 'detailed' repr for devs?

**Comment by TomNicholas** [comment]
> Should we make a 'detailed' repr for devs?

@jbusecke that's what `__repr__` is already supposed to be! The "conventional" thing to do is to have the `__str__` and `__repr__` defined differently, so that `str()` returns a human-readable summary, whereas `repr()` returns something more useful for developers. From [here](https://www.geeksforgeeks.org/str-vs-repr-in-python/):

> "str() is used for creating output for end user while repr() is mainly used for debugging and development. repr’s goal is to be unambiguous and str’s is to be readable. "

Ideally repr actually returns the exact code needed to recreate the object, so that `eval(repr(Grid)) == Grid)`, but when the Grid contains the data for multiple coordinates that isn't very easy. (In really complicated cases like xarray then they just give up and set `repr = str` as good enough basically.)

I notice that `xgcm.Grid` doesn't actually define a `__str__` method at the moment - I think if python `print()` can't find that then it uses repr instead.

Basically if you add a `__str__` method you can have your cake and eat it.
